RestaurantsNearMe/Presentation/SplashScreenViewModel.swift

```diff
@@ -0,0 +1,40 @@
+//
+//  SplashScreenViewModel.swift
+//  RestaurantsNearMe
+//
+//  Created by Colin Evans on 2023-11-24.
+//
+
+import Foundation
+import Combine
+import CloudKit
+
+class SplashScreenViewModel: ObservableObject {
+  @Published var isFetching = false
+  @Published var fetchingError: String?
+  
+  private var cancellables = Set<AnyCancellable>()
+  private let cloudKitService: any CloudKitServiceProviding
+  
+  func fetch() async throws {
+    defer { isFetching = false }
+    isFetching = true
+    do {
+      try await cloudKitService.fetchAPIKeyByID()
+    } catch {
```

Comment:
You can catch different kinds of errors and have them auto-cast:
https://sarunw.com/posts/different-ways-to-catch-throwing-errors-in-swift/#where-clause

Reply:
That's really slick 😁